From 18a1ed3485bacf4efd776ce24e16dd0eca27c66f Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Thu, 29 Mar 2018 15:25:31 +0800
Subject: [PATCH 3550/5242] MLK-17940-2 gpu: imx: imx8_dprc: Specify
 particular PRGs to be blit channels

commit  71ff3c3994ed4c743cdad7ff023284cf20f1890f from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch specifies particular PRGs to be blit channels via
the helper prg_set_blit() if the relevant DPR channel is
a blit channel.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/imx8_dprc.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/gpu/imx/imx8_dprc.c b/drivers/gpu/imx/imx8_dprc.c
index f410df4..79d342d 100644
--- a/drivers/gpu/imx/imx8_dprc.c
+++ b/drivers/gpu/imx/imx8_dprc.c
@@ -747,6 +747,9 @@ static int dprc_probe(struct platform_device *pdev)
 
 		if (i == 1)
 			prg_set_auxiliary(dprc->prgs[i]);
+
+		if (dprc->is_blit_chan)
+			prg_set_blit(dprc->prgs[i]);
 	}
 
 	dprc->dev = dev;
-- 
1.7.9.5

